### PR TITLE
Fix init process callback

### PR DESCRIPTION
**Pull request by dbaluta**
With commit e50ae6be8c7d ("codec_adapter: Initialize processing") the codec adapter does a initialization of processing. This means it needs to process first frames in order to figure out stream parameters. 

Each codec supported by the codec_adapter needs to implement an `.init_process` callback. 

This PR adds the init callbacks for cadence/waves/passthrough codecs and also fixes a problem with passthrough copy() function.

### CODE REVIEW

**Comment by dbaluta** [comment]
This PR depends on https://github.com/thesofproject/sof/pull/3973 and should be merged after.

File: src/audio/codec_adapter/codec/waves.c [hunk 1]
```diff
@@ -648,6 +648,17 @@ int waves_codec_prepare(struct comp_dev *dev)
...
+int waves_codec_init_process(struct comp_dev *dev)
+{
+	struct codec_data *codec = comp_get_codec(dev);
+
+	comp_dbg(dev, "waves_codec_init_process()");
+
+	codec->cpd.produced = 0;
+	codec->cpd.consumed = 0;
+	codec->cpd.init_done = 1;
+}
```
Comment (stolx) [inline src/audio/codec_adapter/codec/waves.c:660]: please add return value

Comment (dbaluta) [reply src/audio/codec_adapter/codec/waves.c:662]: Fixed

**Review by paulstelian97 — APPROVED** [review]

**Review by iuliana-prodan — APPROVED** [review]

**Comment by marc-hb** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
@dbaluta are you able to rebase, this will also re-kick the CI which has had its services restarted.

**Comment by dbaluta** [comment]
@lgirdwood rebased now.

**Review by stolx — APPROVED** [review]
LGTM, thank you
Waves codec works fine with these changes

**Pull request merged**